py/picca/delta_extraction/data_catalogues/desi_healpix.py

```diff
@@ -346,8 +207,181 @@ def read_file(self, filename, catalogue, forests_by_targetid):
 
                 # keep the forest
                 if targetid in forests_by_targetid:
-                    existing_forest = forests_by_targetid[targetid]
-                    existing_forest.coadd(forest)
-                    forests_by_targetid[targetid] = existing_forest
+                    forests_by_targetid[targetid].coadd(forest)
                 else:
                     forests_by_targetid[targetid] = forest
+
+        return forests_by_targetid
+
+    def __call__(self, X):
```

Comment:
Changed the variable name from X to something more explicit. However, it should still be a tuple. That's a limitation of imap -- it can only take one argument.

Reply:
In PR #888 I used `args` to keep using `imap`. We could change it to using `starmap` if we wanted to pass two arguments. I think it's fine either way, so it's up to you really